subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/ReadOnlyFileSystemNode.java

```diff
@@ -16,62 +16,25 @@
 
 package org.gradle.internal.snapshot;
 
-import javax.annotation.Nullable;
-import java.util.Optional;
 import java.util.stream.Stream;
 
 public interface ReadOnlyFileSystemNode {
```

Comment:
`ReadOnlyFileSystemNode` needs a better name name, since it is not any more the read-only part of `FileSystemNode`. This seems to be now a location in the VFS which can have some root snapshots under it. Seems like being a part of the Snapshot hierarchy. How about `SubSnapshotHierarchy` or `SnapshotHierarchySubSet`?

Reply:
The only remaining usage of `ReadOnlyFileSystemNode` is to return an `EMPTY` node. I wonder if we could instead change `FileSystemNode.getNode()` to return an `Optional<FileSystemNode>`.

Reply:
Good point! Done!